### PR TITLE
Add custom tool chat session / message ID dynamic prompting

### PR DESCRIPTION
**Pull request by pablodanswer**
## Description
Fixes Dan-607

https://github.com/danswer-ai/danswer-docs/pull/125

## How Has This Been Tested?
[Describe the tests you ran to verify your changes]

## Accepted Risk
[Any know risks or failure modes to point out to reviewers]

## Related Issue(s)
[If applicable, link to the issue(s) this PR addresses]

## Checklist:
- [ ] All of the automated tests pass
- [ ] All PR comments are addressed and marked resolved
- [ ] If there are migrations, they have been rebased to latest main
- [ ] If there are new dependencies, they are added to the requirements
- [ ] If there are new environment variables, they are added to all of the deployment methods
- [ ] If there are new APIs that don't require auth, they are added to PUBLIC_ENDPOINT_SPECS
- [ ] Docker images build and basic functionalities work
- [ ] Author has done a final read through of the PR right before merge

### CODE REVIEW

File: backend/danswer/tools/custom/custom_tool.py [hunk 1]
```diff
@@ -141,7 +148,10 @@ def run(self, **kwargs: Any) -> Generator[ToolResponse, None, None]:
         request_body = kwargs.get(REQUEST_BODY)
 
         path_params = {}
+
         for path_param_schema in self._method_spec.get_path_param_schemas():
+            print(path_param_schema, kwargs)
```
Comment (Weves) [inline backend/danswer/tools/custom/custom_tool.py]: nit: remove print

**Review by Weves — APPROVED** [review]
seems to be a merge conflict, but other than that + one small nit lgtm

Can we also add a 1-2 sentence guide w/ an example on how to do this somewhere?

**Comment by pablodanswer** [comment]
Added link to docs https://github.com/danswer-ai/danswer-docs/pull/126

**Comment by pablodanswer** [comment]
Screenshot 2024-09-13 at 11 24 23 AM

**Pull request merged**